src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -265,9 +266,15 @@ public function lists($column, $key = null)
      * @param  string  $pageName
      * @param  int|null  $page
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
+     *
+     * @throws InvalidArgumentException
      */
     public function paginate($perPage = null, $columns = ['*'], $pageName = 'page', $page = null)
     {
+        if ($perPage <= 0) {
+            throw new InvalidArgumentException("Negative values can't be used to query results");
```

Comment:
@GrahamCampbell I have looked up all the other open PRs and I found nothing of such, so if `Negative values can't be used to query results` is not good enough, let me know which other one is more appropriate!